sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/InterpretedUnsafeProjection.scala

```diff
@@ -294,7 +294,7 @@ object InterpretedUnsafeProjection {
    */
   @scala.annotation.tailrec
   private def getElementSize(dataType: DataType): Int = dataType match {
-    case NullType | StringType | BinaryType | CalendarIntervalType |
+    case NullType | StringType | BinaryType | CalendarIntervalType | VariantType |
          _: DecimalType | _: StructType | _: ArrayType | _: MapType => 8
```

Comment:
not related to this PR, but I think there are similar code in other places. Shall we unify them and always call `UnsafeRow.isFixedLength`?

Reply:
I agree, that makes sense. From code inspection, one case that I think might produce a different result here is NullType, which I think would instead result in its default size, which is 1. I don't really understand why NullType is being explicitly set to 8 here, so I'm not sure if that would be a problem.

Reply:
Let me know if you want me to make the `UnsafeRow.isFixedLength` change as part of this PR (and if so, what to do about NullType).